assets/prototype/domain/eventEditor/tickets/ticketPriceCalculator/AddPriceModifierButton.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import clone from 'ramda/src/clone';
+import { Button } from '@blueprintjs/core';
+
+/**
+ * Internal imports
+ */
+import { Price } from '../../data/types';
+import { findEntityByGuid } from '../../../shared/predicates/shared/selectionPredicates';
+
+type AddPriceModifierButtonProps = {
+	modifiers: Price[];
+	name: string;
+	price: Price;
+	push: (price: Price) => void;
+	reset: (name: string) => void;
+	sort: () => void;
+};
+
+const AddPriceModifierButton = ({ modifiers, name, price, push, reset, sort }: AddPriceModifierButtonProps) => {
+	const getPriceType = findEntityByGuid(modifiers);
+	return (
+		<Button
+			key={'add'}
+			icon={'add'}
+			onClick={() => {
+				if (price.amount && !isNaN(price.amount)) {
+					const priceClone: Price = clone(price);
+					const baseType: Price = getPriceType(priceClone.priceType);
+					const newPrice = {
+						...priceClone,
+						id: '',
+						isBasePrice: baseType.isBasePrice,
+						isDiscount: baseType.isDiscount,
+						isPercent: baseType.isPercent,
+						isTax: baseType.isTax,
+						order: baseType.order,
+						priceType: baseType.id,
+						priceTypeOrder: baseType.order,
+					};
+					push(newPrice);
+					reset(name);
+					sort();
```

Comment:
On the next iteration, we might consider splitting the logic from here into a logic container.
Also pushing items into `actions` might cause re-renders.

Reply:
that `push()` function is a form mutator for React Final Form and is adding the price into the form data.
I'm not sure if that directly triggers the re-render, but we need to re-render regardless because the new price is added to the list of modifiers for the ticket price